src/main/notifications/index.ts

```diff
@@ -21,146 +21,161 @@ import {NewVersionNotification, UpgradeNotification} from './Upgrade';
-    });
+        download.on('show', () => {
+            flashFrame(true);
+        });
 
-    download.on('click', () => {
-        shell.showItemInFolder(path.normalize());
-    });
-    download.show();
-}
+        download.on('click', () => {
+            shell.showItemInFolder(path.normalize());
+            this.allActiveNotifications.delete(download.uId);
+        });
 
-let upgrade: NewVersionNotification;
+        download.on('close', () => {
+            this.allActiveNotifications.delete(download.uId);
+        });
 
-export function displayUpgrade(version: string, handleUpgrade: () => void): void {
-    if (!Notification.isSupported()) {
-        log.error('notification not supported');
-        return;
-    }
-    if (getDoNotDisturb()) {
-        return;
+        download.on('failed', () => {
+            this.allActiveNotifications.delete(download.uId);
+        });
+        download.show();
     }
 
-    if (upgrade) {
-        upgrade.close();
-    }
-    upgrade = new NewVersionNotification();
-    upgrade.on('click', () => {
-        log.info(`User clicked to upgrade to ${version}`);
-        handleUpgrade();
-    });
-    upgrade.show();
-}
+    public displayUpgrade(version: string, handleUpgrade: () => void): void {
+        if (!Notification.isSupported()) {
+            log.error('notification not supported');
+            return;
+        }
+        if (getDoNotDisturb()) {
+            return;
+        }
 
-let restartToUpgrade;
-export function displayRestartToUpgrade(version: string, handleUpgrade: () => void): void {
-    if (!Notification.isSupported()) {
-        log.error('notification not supported');
-        return;
-    }
-    if (getDoNotDisturb()) {
-        return;
+        if (this.upgradeNotification) {
+            this.upgradeNotification.close();
+        }
+        this.upgradeNotification = new NewVersionNotification();
+        this.upgradeNotification.on('click', () => {
+            log.info(`User clicked to upgrade to ${version}`);
+            handleUpgrade();
+        });
+        this.upgradeNotification.show();
+    }
+
+    public displayRestartToUpgrade(version: string, handleUpgrade: () => void): void {
+        if (!Notification.isSupported()) {
+            log.error('notification not supported');
+            return;
+        }
+        if (getDoNotDisturb()) {
+            return;
+        }
+
+        this.restartToUpgradeNotification = new UpgradeNotification();
+        this.restartToUpgradeNotification.on('click', () => {
+            log.info(`User requested perform the upgrade now to ${version}`);
+            handleUpgrade();
+        });
+        this.restartToUpgradeNotification.show();
     }
 
-    restartToUpgrade = new UpgradeNotification();
-    restartToUpgrade.on('click', () => {
-        log.info(`User requested perform the upgrade now to ${version}`);
-        handleUpgrade();
-    });
-    restartToUpgrade.show();
+    public TEST__getMentionsPerChannel() {
```

Comment:
I'd rather avoid these emergency accessors here, to me I'd say we make an exception for the TS compiler here to allow it to access, or we just use JS for the test ;)